packages/shared/components/atoms/NetworkIcon.svelte

```diff
@@ -4,38 +4,55 @@
     import { NetworkId } from '@core/network'
 
     export let networkId: NetworkId
-    export let height = 22
-    export let width = 22
-    export let outlined = true
+    export let height: number = 22
+    export let width: number = 22
+    export let outlined: boolean = true
+
+    enum Color {
+        Black = 'black',
+        White = 'white',
+        ShimmerHighlight = 'shimmer-highlight',
+        Gray400 = 'gray-400',
+    }
 
     let backgroundColor: string
     let iconColor: string
 
     $: {
         switch (networkId) {
             case NetworkId.Iota:
-                backgroundColor = 'black'
-                iconColor = 'white'
+                backgroundColor = Color.Black
+                iconColor = Color.White
                 break
             case NetworkId.Shimmer:
-                backgroundColor = 'shimmer-highlight'
-                iconColor = 'black'
+                backgroundColor = Color.ShimmerHighlight
+                iconColor = Color.Black
                 break
             case NetworkId.Testnet:
-                backgroundColor = 'gray-400'
-                iconColor = 'black'
+                backgroundColor = Color.Gray400
+                iconColor = Color.Black
                 break
             default:
                 backgroundColor = ''
                 iconColor = ''
         }
     }
+
+    function getBackgroundClasses(color: string): string {
+        return color ? `bg-${color}` : ''
+    }
 </script>
 
-<network-icon
-    class="flex items-center justify-center p-0.5 rounded-full bg-{backgroundColor} {outlined
-        ? 'ring-2 ring-white dark:ring-gray-900'
-        : ''}"
->
+<network-icon class={`${outlined ? 'outlined' : ''} ${getBackgroundClasses(backgroundColor)}`}>
     <Icon {height} {width} icon={NETWORK_ICON_SVG[networkId]} classes="text-{iconColor}" />
 </network-icon>
```

Comment:
I think this component can be refactored a bit more. The enum and the function are unnecessary. You can make a map and use the networkId to index it. 

```svelte
// ...
    $: backgroundColor = classesMap[networkId]?.backgroundColor ?? ''
    $: iconColor = classesMap[networkId]?.iconColor ?? ''

    const classesMap: { [key in NetworkId]?: Record<string, string> } = {
        [NetworkId.Iota]: {
            backgroundColor: 'bg-black',
            iconColor: 'text-white',
        },
        [NetworkId.Shimmer]: {
            backgroundColor: 'bg-shimmer-highlight',
            iconColor: 'text-black',
        },
        [NetworkId.Testnet]: {
            backgroundColor: 'bg-gray-400',
            iconColor: 'text-black',
        },
    }
</script>

<network-icon class={backgroundColor ? backgroundColor : null} class:outlined>
    <Icon {height} {width} icon={NETWORK_ICON_SVG[networkId]} classes={iconColor} />
</network-icon>
```